### ISSUE TITLE
Deferreds not returned in web2 test_wsgi

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2612]
|<img alt="karlb's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| karlb reported|
|-|-|
|Trac ID|trac#2612|
|Type|defect|
|Created|2007-04-30 13:47:54Z|

In twisted/web2/test/test_wsgi.py the deffereds are not properly returned, which let's these tests pass even when they misbehave. The attached patch solves the problem.

Attachments:

* [wsgi_test_missing_return.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/c06/c06ab205f2dcaee5779760276f77e3d95939de82/f8d16b2810eb2f3d0697e20d9fca94cfe8685d14.diff) (2297 bytes) - added by karlb on 2007-04-30 13:52:59Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2612 2612
type__defect defect
reporter__karlb karlb
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web2 web2
keywords__ 
time__1177940874000000 1177940874000000
changetime__1181225614000000 1181225614000000
version__None None
owner__ 
cc__karlb cc__therve cc__exarkun cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

(In [20382]) Return deferred from tests, cleanups.

Refs [#2612](https://github.com/twisted/twisted/issues/2612)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @jyknight|
|-|-|

Ready to review in test-wsgi-deferred-2612.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @therve|
|-|-|

I don't think twisted/web2/test/test_wsgi.py existed in 2001, so the copyright statement probably isn't exactly true.

TestContainer has no docstring

Is the class attribute ```wait_timeout``` on TestContainer used anywhere?  doesn't seem to be

The docstrings that just say "SERVER_PORT" etc are terrible :(  if you can figure out what these tests are supposed to be testing, improving them would be great.  otherwise don't worry about it.

Fixes to the code look great.

**Comment by twisted-trac** [comment]
|<img alt="karlb's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|karlb commented|
|-|-|

The tests check whether the CGI/WSGI environment variables (e.g. SERVER_PORT) are set correctly. Maybe a link to the corresponding part of the WSGI spec should be added?

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve removed owner|
|-|-|

I think exarkun was only talking about the docstrings, not the content of the tests. That should be better now.

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)<a name="note_9"></a>|@jyknight commented|
|-|-|

+1.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_10"></a>|@therve commented|
|-|-|

Is this a review ? :)

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_11"></a>|@ralphm set owner to @therve|
|-|-|

I'll make it one. Changes look fine to me. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_12"></a>|@therve set status to `closed`|
|-|-|

(In [20484]) Merge test-wsgi-deferred-2612

Authors: karlb, therve
Reviewers: ralphm, exarkun
Fixes [#2612](https://github.com/twisted/twisted/issues/2612)

Modify test_wsgi from web2 so that every tests generating Deferreds return
them.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
